packages/headless/src/features/product-recommendations/product-recommendations-actions.ts

```diff
@@ -94,36 +108,51 @@ export const getProductRecommendations = createAsyncThunk<
       return rejectWithValue(fetched.error);
     }
     return {
-      recommendations: fetched.success.results.map(mapResultToProductResult),
+      recommendations: fetched.success.results.map((result) =>
+        mapResultToProductResult(result, {
+          additionalFields: state.productRecommendations.additionalFields || [],
+        })
+      ),
       analyticsAction: logProductRecommendations(),
       searchUid: fetched.success.searchUid,
       duration,
     };
   }
 );
 
-const mapResultToProductResult = (result: Result): ProductRecommendation => {
-  const price = result.raw.ec_price as number | undefined;
-  const promoPrice = result.raw.ec_promo_price as number | undefined;
-  const inStock = result.raw.ec_in_stock as string | undefined;
+const mapResultToProductResult = (
+  result: Result,
+  {additionalFields}: {additionalFields: string[]}
+): ProductRecommendation => {
+  const ec_price = result.raw.ec_price as number | undefined;
+  const ec_promo_price = result.raw.ec_promo_price as number | undefined;
+  const ec_in_stock = result.raw.ec_in_stock as string | undefined;
 
   return {
-    sku: result.raw.permanentid!,
-    name: result.raw.ec_name as string,
-    link: result.clickUri,
-    brand: result.raw.ec_brand as string,
-    category: result.raw.ec_category as string,
-    price,
-    shortDescription: result.raw.ec_shortdesc as string,
-    thumbnailUrl: result.raw.ec_thumbnail as string,
-    imageUrls: result.raw.ec_images as string[],
-    promoPrice:
-      promoPrice === undefined || (price !== undefined && promoPrice >= price)
+    permanentid: result.raw.permanentid!,
+    clickUri: result.clickUri,
+    ec_name: result.raw.ec_name as string,
+    ec_brand: result.raw.ec_brand as string,
+    ec_category: result.raw.ec_category as string,
+    ec_price,
+    ec_shortdesc: result.raw.ec_shortdesc as string,
+    ec_thumbnails: result.raw.ec_thumbnails as string,
+    ec_images: result.raw.ec_images as string[],
+    ec_promo_price:
+      ec_promo_price === undefined ||
+      (ec_price !== undefined && ec_promo_price >= ec_price)
+        ? undefined
+        : ec_promo_price,
+    ec_in_stock:
+      ec_in_stock === undefined
         ? undefined
-        : promoPrice,
-    inStock:
-      inStock === undefined ? undefined : inStock.toLowerCase() === 'yes',
-    rating: result.raw.ec_rating as number,
+        : ec_in_stock.toLowerCase() === 'yes' ||
+          ec_in_stock.toLowerCase() === 'true',
+    ec_rating: result.raw.ec_rating as number,
+    additionalFields: additionalFields.reduce(
+      (all, field) => ({...all, [field]: result.raw[field]}),
```

Comment:
Wondering if I should put `result[field] || result.raw[field]` here, any opinion?

Reply:
At face value, I think looking at both top level properties and "custom" fields makes sense, yes.

But it's also up to you to determine the "correct" usage/what people are trying to do with this controller at the end of the day, to power their commerce deployment, so you and your team opinion matters more than mine ;)

Reply:
I do not understand the question, aren't the additional fields only available on the `raw` object?
Wouldn't `result.raw[field]` be the only option?

Reply:
`title`, `excerpt`, `summary`, `childResults`, those are fields that might be required

Reply:
Ok, I see, would it make sense to just expose the entire result under a `result` property?

The controller would give first class support to the fields we expected 90% of people to use, but there is still an escape hatch to access other fields.